chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/CertificationResultManager.java

```diff
@@ -543,6 +550,74 @@ private int updateUcdProcesses(CertificationResult certResult, List<UcdProcess>
+                // existing listing has none, add all from the update
+                for (CertificationResultConformanceMethod updatedItem : updatedConformanceMethods) {
+                    conformanceMethodToAdd.add(updatedItem);
+                }
+            } else if (existingConformanceMethods.size() > 0) {
+                // existing listing has some, compare to the update to see if
+                // any need to be added
+                for (CertificationResultConformanceMethod updatedItem : updatedConformanceMethods) {
+                    boolean inExistingListing = false;
+                    for (CertificationResultConformanceMethod existingItem : existingConformanceMethods) {
+                        if (updatedItem.matches(existingItem)) {
+                            inExistingListing = true;
+                        }
+                    }
+
+                    if (!inExistingListing) {
+                        conformanceMethodToAdd.add(updatedItem);
+                    }
+                }
+            }
+        }
+
+        // figure out which data to remove
+        if (existingConformanceMethods != null && existingConformanceMethods.size() > 0) {
+            // if the updated listing has none, remove them all from existing
+            if (updatedConformanceMethods == null || updatedConformanceMethods.size() == 0) {
+                for (CertificationResultConformanceMethod existingItem : existingConformanceMethods) {
+                    idsToRemove.add(existingItem.getId());
+                }
+            } else if (updatedConformanceMethods.size() > 0) {
+                for (CertificationResultConformanceMethod existingItem : existingConformanceMethods) {
+                    boolean inUpdatedListing = false;
+                    for (CertificationResultConformanceMethod updatedItem : updatedConformanceMethods) {
+                        inUpdatedListing = !inUpdatedListing ? existingItem.matches(updatedItem) : inUpdatedListing;
+                    }
+                    if (!inUpdatedListing) {
+                        idsToRemove.add(existingItem.getId());
+                    }
+                }
+            }
+        }
+
+        numChanges = conformanceMethodToAdd.size() + idsToRemove.size();
+        for (CertificationResultConformanceMethod toAdd : conformanceMethodToAdd) {
+            CertificationResultConformanceMethodEntity toAddEntity = new CertificationResultConformanceMethodEntity();
+            toAddEntity.setCertificationResultId(certResult.getId());
+            toAddEntity.setConformanceMethodId(toAdd.getConformanceMethod().getId());
+            toAddEntity.setVersion(toAdd.getConformanceMethodVersion());
+            certResultDAO.addConformanceMethodMapping(toAddEntity);
+        }
+
+        for (Long idToRemove : idsToRemove) {
+            certResultDAO.deleteConformanceMethodMapping(idToRemove);
+        }
+        return numChanges;
+    }
+
```

Comment:
I'm not a huge fan of this logic - I know it's used all over in the API code.  I have been trying to use "subtraction" of the lists to determine what has been added and removed.  Maybe it doesn't apply here - I'm not 100% sure.  This is definitely up for discussion...
`updated - existing = added items`
`existing - updated = removed items`
There is a working example in `CertificationResultManager` around line 1272.
![image](https://user-images.githubusercontent.com/24817202/135876766-2de99307-7ffb-4c64-bb4c-9cb95ac70261.png)